src/mlpack/methods/ann/layer/sampling_impl.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file sampling_impl.hpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Sampling class which samples from parameters for a given
+ * distribution.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SAMPLING_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SAMPLING_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "sampling.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+Sampling<InputDataType, OutputDataType>::Sampling()
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+Sampling<InputDataType, OutputDataType>::Sampling(
+    const size_t inSize,
+    const size_t outSize) :
+    inSize(inSize),
+    outSize(outSize)
+{
+  if (inSize != 2 * outSize)
+  {
+    Log::Fatal << "The input size of Sampling layer should be 2 * output size!"
+        << std::endl;
+  }
+}
+
+template <typename InputDataType, typename OutputDataType>
+Sampling<InputDataType, OutputDataType>::Sampling(
+    const size_t sampleSize) :
+    outSize(sampleSize)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void Sampling<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, arma::Mat<eT>&& output)
+{
+  if (input.n_rows != 2 * outSize)
+  {
+    Log::Fatal << "The output size of layer before the Sampling layer should "
+        << "be 2 * output size of the Sampling layer!" << std::endl;
+  }
+
+  arma::arma_rng::set_seed_random();
```

Comment:
Setting the seed inside core code should be avoided. This way we could test the 'Forward' function by putting a constant seed in test function. Its user's job to set the seed properly.

Reply:
Just FYI, if you do want to set the seed, we have a function `math::RandomSeed()` that will set it for both mlpack and Armadillo.

A bit more information on random seeds---if the user never calls `math::RandomSeed()` or equivalent, then the program that is compiled will have one hard-coded seed that's set at compile time.  A lot of the command-line programs have an option to set the random seed with the `--seed` option, or alternately they will call `math::RandomSeed(std::time(NULL))` to choose a random seed based on the time.